### DISCUSSION TITLE
Strict-Server Isn't Very Strict

### DISCUSSION THREAD

**Discussion by spasarok** [oapi-codegen/oapi-codegen#906] [Ideas]
I've been messing around with the `strict-server` capability using a Gin server and it's not working quite as I'd expect.

Here are the capabilities I see:

- `strict-server` **does** enforce content type
- `strict-server` **does** enforce parameter types

Here are the capabilities I don't see:

- `strict-server` **does not** enforce required body parameters
- `strict-server` **does not** enforce regex patterns for string parameters
- `strict-server` **does not** return a 400 even if the request body contains additional random keys
- `strict-server` **does not** return an error message to the user and does not provide a way to add middleware to do so

People could write their own middleware to do some of these validations, but since the constraints are already expressed in the API spec it would be nice if the tool could generate the middleware to enforce those constraints. Are any of these features on the roadmap? Thanks!

**Comment by jamietanna** [comment]
Hey, so for some of these, they can be performed using the `OapiRequestValidator` middleware - we intentionally don't implement all of these checks and instead leverage other libraries for them.
